File: app/src/main/java/com/awscherb/cardkeeper/ui/base/BaseApplication.java
package com.awscherb.cardkeeper.ui.base;

import android.app.Application;

import com.awscherb.cardkeeper.di.component.DaggerViewComponent;
import com.awscherb.cardkeeper.di.component.ViewComponent;
import com.awscherb.cardkeeper.di.module.AppModule;
import com.awscherb.cardkeeper.di.module.DaoModule;

public class BaseApplication extends Application {

    private ViewComponent viewComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        viewComponent = DaggerViewComponent.builder()
                .appModule(new AppModule(this))
                .daoModule(new DaoModule())
                .build();

    }

    public ViewComponent getViewComponent() {
        return viewComponent;
    }


    @Override
    public void onTerminate() {
        super.onTerminate();
    }
}

File: app/src/main/java/com/awscherb/cardkeeper/ui/base/BaseApplication.kt
package com.awscherb.cardkeeper.ui.base

import android.app.Application

import com.awscherb.cardkeeper.di.component.DaggerViewComponent
import com.awscherb.cardkeeper.di.component.ViewComponent
import com.awscherb.cardkeeper.di.module.AppModule
import com.awscherb.cardkeeper.di.module.DaoModule
import com.awscherb.cardkeeper.di.module.SchedulerModule

class BaseApplication : Application() {

    lateinit var viewComponent: ViewComponent
        private set

    override fun onCreate() {
        super.onCreate()

        viewComponent = DaggerViewComponent.builder()
                .appModule(AppModule(this))
                .daoModule(DaoModule())
                .schedulerModule(SchedulerModule())
                .build()

    }
}
